examples/bounding_box.cu

```diff
@@ -66,21 +66,17 @@ int main(void)
   thrust::device_vector<point2d> points(N);
   
   // generate some random points in the unit square
-  for(size_t i = 0; i < N; i++)
-  {
-      float x = u01(rng);
-      float y = u01(rng);
-      points[i] = point2d(x,y);
-  }
+  std::generate(points.begin(), points.end(), [&] {
+    auto x = u01(rng);
+    auto y = u01(rng);
```

Comment:
Let's avoid auto here, we're not requiring C++11 yet.

Reply:
If we're not requiring C++11, we can't use this lambda either, can we?